File: DP/python/9465.py
result = []
T = int(input())
for _ in range(T):
    n = int(input())
    data = [list(map(int,input().split())) for _ in range(2)]
    if n>=2:
        data[0][1] += data[1][0]
        data[1][1] += data[0][0]
    for i in range(2,n):
        data[0][i] += max(data[1][i-1],data[1][i-2])
        data[1][i] += max(data[0][i-1],data[0][i-2])
    result.append(max(data[0][-1],data[1][-1]))
print("\n".join(map(str,result)))

File: DP/javascript/9465.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');

let result = "";
for (let i = 2 ; i<input.length-1 ; i+=3){
    const dp = [];
    dp.push(input[i].split(' ').map(Number)); // 1행
    dp.push(input[i+1].split(' ').map(Number)); // 2행
    if (dp[0].length >= 2) {
        dp[0][1] += dp[1][0];
        dp[1][1] += dp[0][0];
    }
    for (let j = 2 ; j<dp[0].length ; j++ ){
        dp[0][j] += Math.max(dp[1][j-1],dp[1][j-2]);
        dp[1][j] += Math.max(dp[0][j-1],dp[0][j-2]);
    }
    result += Math.max(dp[0][dp[0].length-1],dp[1][dp[1].length-1]) + "\n";
}
console.log(result);